### PR TITLE
nogo: plumb SDK Go version into go/types

### PR DESCRIPTION
**Pull request by sluongng**
Forward the selected SDK Go version from compilepkg into nogo and initialize the corresponding go/types version plumbing during type checking. This keeps nogo aligned with the active toolchain version without keying the action on unrelated gcflags.

Keep gc_goopts -lang overrides unsupported for now. RunNogo still forwards only the SDK version, and compilepkg now documents that -lang parsing may be added later once nogo has proper gc flag parsing.

Add focused nogo coverage that checks both the RunNogo action

### CODE REVIEW

**Comment by sluongng** [comment]
I played around with supporting `-lang` for a bit and found the effort to be not worth it (need to filter gc_goopts for -lang values in starlark, then passed it down, etc...)

If somebody found a use case for it in the future, feel free to add.

File: go/private/actions/compilepkg.bzl [hunk 1]
```diff
@@ -216,6 +221,7 @@ def emit_compilepkg(
             sources = sources,
             cgo_go_srcs = cgo_go_srcs,
             archives = archives,
+            go_version = _sdk_go_version(go.sdk.version),
```
Comment (fmeum) [inline go/private/actions/compilepkg.bzl]: This retains a new string per action. Either store the new string in the SDK providers or make the transformation lazy via `Args`.

Comment (sluongng) [reply go/private/actions/compilepkg.bzl]: We can probably just pass down the value as-is and append the prefix with nogo main

**Comment by sluongng** [comment]
@fmeum AI code review turned out to be pretty useful here. It called out for some edge cases that is required to support earlier Go versions, as well as some cases involve rc releases, or dev releases built from a commit hash. I think the attached tests act as pretty good documentation for those cases though.

PTAL

**Comment by fmeum** [comment]
@sluongng I don't see those changes, did you push them?

**Comment by sluongng** [comment]
Duh i pushed to the wrong branch. Give me a bit

**Review by jayconrod — APPROVED** [review]
